src/testing/adaptive_sampling_testing.cpp

```diff
@@ -0,0 +1,167 @@
+        flow_solver->ode_solver->steady_state();
+
+        dealii::LinearAlgebra::distributed::Vector<double> standard_solution(flow_solver->dg->solution);
+        dealii::LinearAlgebra::distributed::Vector<double> implicit_solution(flow_solver_implicit->dg->solution);
+
+        double standard_error = ((standard_solution-=implicit_solution).l2_norm()/implicit_solution.l2_norm());
+        double standard_func_error = functional->evaluate_functional(false,false) - functional_implicit->evaluate_functional(false,false);
+
+        pcout << "State error: " << std::setprecision(15) << standard_error << std::setprecision(6) << std::endl;
+        pcout << "Functional error: " << std::setprecision(15) << standard_func_error << std::setprecision(6) << std::endl;
+
+        data_table->add_value("Parameter 1", parameter(0));
+        data_table->add_value("Parameter 2", parameter(1));
+        data_table->add_value("FOM func", functional_implicit->evaluate_functional(false,false));
+        data_table->add_value("ROM func", functional->evaluate_functional(false,false));
+        data_table->add_value("Func error", standard_func_error);
+        data_table->add_value("State error", standard_error);
+
+        data_table->set_precision("Parameter 1", 16);
+        data_table->set_precision("Parameter 2", 16);
+        data_table->set_precision("FOM func", 16);
+        data_table->set_precision("ROM func", 16);
+        data_table->set_precision("Func error", 16);
+        data_table->set_precision("State error", 16);
+
+        std::ofstream data_table_file("adaptation_data_table.txt");
+        data_table->write_text(data_table_file, dealii::TableHandler::TextOutputFormat::org_mode_table);
+    }
+    return 0;
+}
+
+template <int dim, int nstate>
+Parameters::AllParameters AdaptiveSamplingTesting<dim, nstate>::reinitParams(RowVector2d parameter) const{
+    // Copy all parameters
+    PHiLiP::Parameters::AllParameters parameters = *(this->all_parameters);
+
+    using FlowCaseEnum = Parameters::FlowSolverParam::FlowCaseType;
+    const FlowCaseEnum flow_type = this->all_parameters->flow_solver_param.flow_case_type;
+    if (flow_type == FlowCaseEnum::burgers_rewienski_snapshot){
+        parameters.burgers_param.rewienski_a = parameter(0);
+        parameters.burgers_param.rewienski_b = parameter(1);
+    }
+    else if (flow_type == FlowCaseEnum::naca0012){
+        //const double pi = atan(1.0) * 4.0;
+        parameters.euler_param.mach_inf = parameter(0);
+        parameters.euler_param.angle_of_attack = parameter(1); //Already in radians
+    }
+    else{
+        this->pcout << "Invalid flow case. You probably forgot to specify a flow case in the prm file." << std::endl;
+        std::abort();
+    }
+    return parameters;
+}
+
+template <int dim, int nstate>
+std::shared_ptr<Functional<dim,nstate,double>> AdaptiveSamplingTesting<dim, nstate>::functionalFactory(std::shared_ptr<DGBase<dim, double>> dg) const
```

Comment:
To be moved later on. Can you open an issue about that.

Reply:
I just realized that there is a functional factory class in functional.cpp. I therefore removed this function and I'm now using the functional factory class. I added lift, drag and solution_integral (for Burgers Rewienski) to the factory, see most recent commit.